editor/src/clj/editor/settings_core.clj

```diff
@@ -38,22 +55,26 @@
   (BufferedReader. (StringReader. content)))
 
 (defn- empty-parse-state []
-  {:current-category nil :settings nil})
+  {:current-category nil :settings []})
 
 (defn- parse-category-line [{:keys [current-category settings] :as parse-state} line]
   (when-let [[_ new-category] (re-find #"^\s*\[([^\]]*)\]" line)]
     (assoc parse-state :current-category new-category)))
 
-(defn- parse-setting-line [{:keys [current-category settings] :as parse-state} line]
-  (when-let [[_ key val] (seq (map s/trim (re-find #"([^=]+)=(.*)" line)))]
-    (when-let [setting-path (seq (non-blank (s/split key #"\.")))]
-      (update parse-state :settings conj {:path (into [current-category] setting-path) :value val}))))
+ (defn- parse-setting-line [{:keys [current-category settings] :as parse-state} line]
```

Comment:
Here we parse a line read from game.project. We now either set the value as before if the key doesn't already exist or we update with a concatenated value.